MREUnityRuntime/MREUnityRuntimeLib/Behaviors/Handlers/PenHandler.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+using MixedRealityExtension.API;
+using MixedRealityExtension.App;
+using MixedRealityExtension.Behaviors.ActionData;
+using MixedRealityExtension.Behaviors.Actions;
+using MixedRealityExtension.Core.Interfaces;
+using MixedRealityExtension.PluginInterfaces.Behaviors;
+using System;
+
+namespace MixedRealityExtension.Behaviors.Handlers
+{
+	internal class PenToolHandler : ToolHandler<PenToolData>
+	{
+		internal PenToolHandler(IPenBehavior tool, WeakReference<MixedRealityExtensionApp> appRef, IActor attachedActor)
+			: base(tool, appRef, attachedActor)
+		{
+			
+		}
+
+		internal static PenToolHandler Create(IActor actor, WeakReference<MixedRealityExtensionApp> appRef)
+		{
+			var behaviorFactory = MREAPI.AppsAPI.BehaviorFactory;
+			var penBehavior = behaviorFactory.GetOrCreatePenBehavior(actor);
```

Comment:
>GetOrCreatePenBehavior [](start = 37, length = 22)

You're calling into application code here. It's good practice to protect against invalid return values and thrown exceptions.

Reply:
Yeah.  We hadn't done that elsewhere in the behavior system, but can add that in now.